ios/authentication.md

```diff
@@ -1,344 +1,417 @@
 
-2. Add the following dependencies in the Podfile.
+```swift
+AWSMobileClient.sharedInstance().getTokens { (tokens, error) in
+    if let error = error {
+        print("Error getting token \(error.localizedDescription)")
+    } else if let tokens = tokens {
+        print(tokens.accessToken!.tokenString!)
+    }
+}
+```
 
-	```ruby
-	platform :ios, '9.0'
-	target :'YOUR-APP-NAME' do
-	  use_frameworks!
-	  pod 'AWSMobileClient', '~> 2.6.33'
-	  pod 'AWSGoogleSignIn', '~> 2.6.33'
-	  pod 'AWSUserPoolsSignIn', '~> 2.6.33'
-	  pod 'AWSAuthUI', '~> 2.6.33'
-	  pod 'GoogleSignIn', '~> 4.0'
-	  # other pods
-	end
-	```
-	Run `pod install --repo-update` before you continue.
+#### OIDC Tokens
 
-3. Add Google metadata to `Info.plist`.
+```swift
+AWSMobileClient.sharedInstance().getTokens { (tokens, error) in
+    if let error = error {
+        print("Error getting token \(error.localizedDescription)")
+    } else if let tokens = tokens {
+        print(tokens.accessToken!.tokenString!)
+    }
+}
+```
 
-	To configure your Xcode project to use Google Login, open its `Info.plist` file using **Right-click > Open As > Source Code.** Add the following entry. Substitute your project name for the placeholder string.
+#### AWS Credentials
 
-	```xml
+```swift
+AWSMobileClient.sharedInstance().getAWSCredentials { (credentials, error) in
+    if let error = error {
+        print("\(error.localizedDescription)")
+    } else if let credentials = credentials {
+        print(credentials.accessKey)
+    }
+}
+```
 
-	<plist version="1.0">
-	<!-- YOUR OTHER PLIST ENTRIES HERE -->
+## Federated Identities (Social Sign-in)
 
-	<!-- ADD AN ENTRY TO CFBundleURLTypes for Google -->
-	<!-- IF YOU DO NOT HAVE CFBundleURLTypes, YOU CAN COPY THE WHOLE BLOCK BELOW -->
-	<key>CFBundleURLTypes</key>
-	<array>
-	    <dict>
-	    <key>CFBundleURLSchemes</key>
-	    <array>
-	        <string>com.googleusercontent.apps.xxxxxxxxxxxx-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx</string>
-	    </array>
-	    </dict>
-	</array>
+**Availability Note**
+Currently, the federation feature in the AWSMobileClient supports Cognito Identity Pools only.
+{: .callout .callout--info}
 
-	<!-- ... -->
-	```
+### Federated Sign In
 
-## Set Up Facebook Authentication
+```swift
+AWSMobileClient.sharedInstance().federatedSignIn(providerName: "graph.facebook.com", token: "FACEBOOK_TOKEN_HERE") { (userState, err)  in
+    if let error = error {
+        print("Federated Sign In failed: \(error.localizedDescription)")
+    }
+}
+```
+
+`federatedSignIn` can be used to obtain federated "Identity ID" using external providers like Google, Facebook or Twitter. If the tokens are expired and new tokens are needed, a notification will be dispatched on the `AWSMobileClient` listener with the user state `signedOutFederationTokensInvalid`. You can give the updated tokens via the same `federatedSignIn` method. 
```

Comment:
federatedSignIn() - be consistent in marking methods as such

Reply:
same goes for signedOutFederationTokensInvalid()

Reply:
Changed for the federatedSignIn() however signedOutFederationTokensInvalid is a dispatched state and not a method.